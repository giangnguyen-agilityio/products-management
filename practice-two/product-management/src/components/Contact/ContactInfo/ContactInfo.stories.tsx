import type { Meta, StoryObj } from '@storybook/react'
import ContactInfo from './index'

const meta = {
  title: 'Practice Two/ContactInfo Component',
  component: ContactInfo,
  decorators: [
    Story => (
      <div
        style={{
          backgroundColor: '#ECB203',
          width: '100vw',
          height: '100%',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof ContactInfo>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {}
