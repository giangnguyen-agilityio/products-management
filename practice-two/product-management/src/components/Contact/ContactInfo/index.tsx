import { memo } from 'react'
import { Container, Grid, Text } from '@chakra-ui/react'
import { Link } from 'react-router-dom'
import ContactInfoItem from '../ContactInfoItem'

// Import icons and assets
import locationIcon from '@assets/icons/Location_Icon.svg'
import telephoneIcon from '@assets/icons/Telephone_Icon.svg'
import faxIcon from '@assets/icons/Fax_Icon.svg'
import emailIcon from '@assets/icons/Email_Icon.svg'

// ContactInfo component
const ContactInfo = () => {
  return (
    <Container as="section" className="contact-information" padding={0}>
      <Grid
        gap="15px"
        margin="-150px auto 0"
        gridTemplateColumns={{
          base: '1fr',
          md: 'repeat(2, 1fr)',
          lg: 'repeat(4, 1fr)',
        }}
      >
        {/* ContactInfoItem for main office location */}
        <ContactInfoItem
          icon={locationIcon}
          title="our main office"
          content="SoHo 94 Broadway St New York, NY 1001"
        />
        {/* ContactInfoItem for phone numbers */}
        <ContactInfoItem
          icon={telephoneIcon}
          title="phone number"
          content={
            <>
              <Text>234-9876-5400</Text>
              <Text marginTop="20px">888-0123-4567 (Toll Free)</Text>
            </>
          }
        />
        {/* ContactInfoItem for fax */}
        <ContactInfoItem icon={faxIcon} title="fax" content="1-234-567-8900" />
        {/* ContactInfoItem for email */}
        <ContactInfoItem
          icon={emailIcon}
          title="email"
          content={
            <Link to="/">
              <Text
                as="span"
                color="primary"
                variant="fontPrimaryBold"
                textDecoration="unset"
                _hover={{
                  textDecoration: 'underline',
                }}
              >
                hello@theme.com
              </Text>
            </Link>
          }
        />
      </Grid>
    </Container>
  )
}

export default memo(ContactInfo)
