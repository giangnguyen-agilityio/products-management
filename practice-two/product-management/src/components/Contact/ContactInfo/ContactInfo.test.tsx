import { render } from '@testing-library/react'
import ContactInfo from '.'

jest.mock('@assets/icons/Location_Icon.svg', () => ({
  default: 'locationIcon',
}))

jest.mock('@assets/icons/Telephone_Icon.svg', () => ({
  default: 'telephoneIcon',
}))

jest.mock('@assets/icons/Fax_Icon.svg', () => ({
  default: 'faxIcon',
}))

jest.mock('@assets/icons/Email_Icon.svg', () => ({
  default: 'emailIcon',
}))

describe('ContactInfo component', () => {
  it('renders contact information', () => {
    const { container, getByText } = render(<ContactInfo />)

    expect(getByText('our main office')).toBeInTheDocument()
    expect(getByText('phone number')).toBeInTheDocument()
    expect(getByText('fax')).toBeInTheDocument()
    expect(getByText('email')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
