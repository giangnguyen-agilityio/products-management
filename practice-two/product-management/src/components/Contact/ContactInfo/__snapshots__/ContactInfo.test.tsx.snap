// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo component renders contact information 1`] = `
<div>
  <section
    class="chakra-container contact-information css-old1by"
  >
    <div
      class="css-1w32bio"
    >
      <div
        class="css-1gfsowa"
      >
        <div
          class="css-wkpw2c"
        >
          <img
            alt="our main office icon"
            class="chakra-image css-1wsuzl7"
            src="[object Object]"
          />
          <p
            class="chakra-text css-31u82y"
          >
            our main office
          </p>
        </div>
        SoHo 94 Broadway St New York, NY 1001
      </div>
      <div
        class="css-1gfsowa"
      >
        <div
          class="css-wkpw2c"
        >
          <img
            alt="phone number icon"
            class="chakra-image css-1wsuzl7"
            src="[object Object]"
          />
          <p
            class="chakra-text css-31u82y"
          >
            phone number
          </p>
        </div>
        <p
          class="chakra-text css-0"
        >
          234-9876-5400
        </p>
        <p
          class="chakra-text css-1xaekgw"
        >
          888-0123-4567 (Toll Free)
        </p>
      </div>
      <div
        class="css-1gfsowa"
      >
        <div
          class="css-wkpw2c"
        >
          <img
            alt="fax icon"
            class="chakra-image css-1wsuzl7"
            src="[object Object]"
          />
          <p
            class="chakra-text css-31u82y"
          >
            fax
          </p>
        </div>
        1-234-567-8900
      </div>
      <div
        class="css-1gfsowa"
      >
        <div
          class="css-wkpw2c"
        >
          <img
            alt="email icon"
            class="chakra-image css-1wsuzl7"
            src="[object Object]"
          />
          <p
            class="chakra-text css-31u82y"
          >
            email
          </p>
        </div>
        <a
          href="/"
        >
          <span
            class="chakra-text css-t8ai10"
          >
            hello@theme.com
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;
